### PR TITLE
Bug 1797244: test/extended/dr: update test to reflect new DR process

### PR DESCRIPTION
**Pull request by hexfusion**
The addition of cluster etcd operator introduced a new disaster recovery workflow that broke the disruptive tests. The previous workflow involved doing a snapshot restore of the etcd data file. The advantage to this approach is that once etcd starts the process is complete. The downside is a restore action must take place on each master node.

In 4.4+ we automate this process and require only a single invocation of the cluster-backup.sh script and cluster-restore.sh script. Given new machines the operator then scales etcd on the new machines resulting in a 3 node etcd cluster.

The downside of this process is that it is more disruptive as etcd and the rest of the control plane scales. For this reason, we needed to adjust not only some of the timings of the old test allowing for certain steps to take longer. But also to allow for transient etcd timeouts. During scaling from 1 to 2 etcd members we must lose quorum. The resulting leader election will result in client timeouts.

This is a starting place where we can begin to harden the requirements and improve. But for now lets make sure the workflow itself completes as expected.

### CODE REVIEW

**Comment by hexfusion** [comment]
flake 
/test e2e-aws-disruptive

**Comment by hexfusion** [comment]
> [sig-etcd][Feature:DisasterRecovery][Disruptive] [Feature:EtcdRecovery] Cluster should restore itself after quorum loss [Disabled:Broken] [Serial] [Suite:openshift]

Passed \o/

test failed because of what I thought was https://github.com/openshift/origin/pull/25087 but CI merged already

File: test/extended/dr/quorum_restore.go [hunk 1]
```diff
@@ -229,20 +233,20 @@ var _ = g.Describe("[sig-etcd][Feature:DisasterRecovery][Disruptive]", func() {
...
 				o.Expect(err).NotTo(o.HaveOccurred())
 
 				framework.Logf("Force new revision of kube-apiserver")
-				_, err = oc.AdminOperatorClient().OperatorV1().KubeAPIServers().Patch(context.Background(), "cluster", types.MergePatchType, []byte(`{"spec": {"forceRedepl  oymentReason": "recover-kube-apiserver"}}`), metav1.PatchOptions{})
+				_, err = oc.AdminOperatorClient().OperatorV1().KubeAPIServers().Patch(context.Background(), "cluster", types.MergePatchType, []byte(`{"spec": {"forceRedeploymentReason": "recover-kube-apiserver"}}`), metav1.PatchOptions{})
 				o.Expect(err).NotTo(o.HaveOccurred())
 
 				framework.Logf("Wait for etcd pods to become available")
-				_, err = exutil.WaitForPods(
+				_, err = waitForPodsTolerateClientTimeout(
```
Comment (smarterclayton) [inline test/extended/dr/quorum_restore.go:244]: wait, why do you have to tolerate timeouts here?  at this point etcd should be available?

File: test/extended/dr/quorum_restore.go [hunk 2]
```diff
@@ -242,7 +242,7 @@ var _ = g.Describe("[sig-etcd][Feature:DisasterRecovery][Disruptive]", func() {
 					exutil.ParseLabelsOrDie("k8s-app=etcd"),
 					exutil.CheckPodIsReady,
 					expectedNumberOfMasters,
-					20*time.Minute,
+					80*time.Minute,
```
Comment (smarterclayton) [inline test/extended/dr/quorum_restore.go]: wow... why?

File: test/extended/dr/quorum_restore.go [hunk 3]
```diff
@@ -147,9 +142,9 @@ var _ = g.Describe("[sig-etcd][Feature:DisasterRecovery][Disruptive]", func() {
 				}
 
 				framework.Logf("Perform etcd backup on remaining machine %s (machine %s)", survivingNodeName, survivingMachineName)
-				expectSSH("sudo -i /bin/bash -c 'rm -f /root/assets/backup/snapshot*'; sudo -i /bin/bash -x /usr/local/bin/etcd-snapshot-backup.sh /root/assets/backup", survivingNode)
-				framework.Logf("Restore etcd on remaining node %s (machine %s)", survivingNodeName, survivingMachineName)
-				expectSSH(fmt.Sprintf("sudo -i /bin/bash -c '/bin/bash -x /usr/local/bin/etcd-snapshot-restore.sh /root/assets/backup/snapshot* %s'", etcdConnectionString), survivingNode)
+				expectSSH("sudo -i /bin/bash -cx 'sudo rm -rf /home/core/backup; sudo /usr/local/bin/cluster-backup.sh ~core/backup'", survivingNode)
```
Comment (smarterclayton) [inline test/extended/dr/quorum_restore.go:148]: why do you have to sudo when you're running in sudo?

File: test/extended/dr/quorum_restore.go [hunk 4]
```diff
@@ -211,15 +206,17 @@ var _ = g.Describe("[sig-etcd][Feature:DisasterRecovery][Disruptive]", func() {
...
 						defer func() {
 							if r := recover(); r != nil {
 								fmt.Println("Recovered from panic", r)
 							}
 						}()
 						nodes, err := oc.AdminKubeClient().CoreV1().Nodes().List(context.Background(), metav1.ListOptions{LabelSelector: "node-role.kubernetes.io/master="})
 						if err != nil {
-							return false, err
+							// scale up to 2nd etcd will make this failure inevitable
+							framework.Logf("return false - err %v", err)
```
Comment (smarterclayton) [inline test/extended/dr/quorum_restore.go]: can you improve this error message to say "ignoring errors while waiting for masters to recover: %v"

File: test/extended/dr/quorum_restore.go (hunk 2 above)
Comment (hexfusion) [reply test/extended/dr/quorum_restore.go]: needs tuning I am verifying those now.

File: test/extended/dr/quorum_restore.go (hunk 1 above)
Comment (hexfusion) [reply test/extended/dr/quorum_restore.go:244]: I am forcing a revision of both etcd and apiserver static pod resources. This causes disruption as we transfer leadership.  I will try to dial this in but it has proven a challenge. The error comes from apiserver.

File: test/extended/dr/quorum_restore.go (hunk 3 above)
Comment (hexfusion) [reply test/extended/dr/quorum_restore.go:148]: typo, thanks

File: test/extended/dr/quorum_restore.go (hunk 4 above)
Comment (hexfusion) [reply test/extended/dr/quorum_restore.go]: sure thing

Comment (hexfusion) [reply test/extended/dr/quorum_restore.go]: hmm rebase is off here let me verify...

**Pull request merged**